extension/main.js

```diff
@@ -1,210 +1,91 @@
+			TabsService.openTab(tabUrl, tab);
+		});
+	} else {
+		TabsService.openTab(tabUrl, tab);
 	}
+}
 
-	function scheduleAlarm(period) {
-		// unconditionally schedule alarm
-		// period is in minutes
-		chrome.alarms.create({when: Date.now() + 2000 + (period * 60 * 1000)});
+function handleInstalled(details) {
+	if (details.reason === 'install') {
+		chrome.runtime.openOptionsPage();
 	}
+}
 
-	function update() {
-		window.gitHubNotifCount().then(response => {
-			const count = response.count;
-			const interval = response.interval;
-			const lastModifed = response.lastModifed;
-			const period = handleInterval(interval);
-
-			scheduleAlarm(period);
-			handleLastModified(lastModifed);
+chrome.alarms.create({when: Date.now() + 2000});
+chrome.alarms.onAlarm.addListener(update);
+chrome.runtime.onMessage.addListener(update);
 
-			render(handleCount(count), [65, 131, 196, 255], 'Notifier for GitHub');
-		}).catch(handleError);
-	}
+PermissionsService.queryPermission('notifications').then(granted => {
+	if (granted) {
+		chrome.notifications.onClicked.addListener(id => {
+			NotificationsService.openNotification(id);
+		});
 
-	function openTab(url, tab) {
-		// checks optional permissions
-		window.GitHubNotify.queryPermission('tabs').then(granted => {
-			if (granted) {
-				const currentWindow = true;
-				chrome.tabs.query({currentWindow, url}, tabs => {
-					if (tabs.length > 0) {
-						const highlighted = true;
-						chrome.tabs.update(tabs[0].id, {highlighted, url});
-					} else if (tab && tab.url === 'chrome://newtab/') {
-						chrome.tabs.update(null, {url});
-					} else {
-						chrome.tabs.create({url});
-					}
-				});
-			} else {
-				chrome.tabs.create({url});
-			}
+		chrome.notifications.onClosed.addListener(id => {
+			NotificationsService.removeNotification(id);
 		});
 	}
+});
 
-	chrome.alarms.create({when: Date.now() + 2000});
-	chrome.alarms.onAlarm.addListener(update);
-	chrome.runtime.onMessage.addListener(update);
-
-	window.GitHubNotify.queryPermission('notifications').then(granted => {
-		if (granted) {
-			chrome.notifications.onClicked.addListener(handleNotificationClicked);
-			chrome.notifications.onClosed.addListener(handleNotificationClosed);
-		}
-	});
-
-	// launch options page on first run
-	chrome.runtime.onInstalled.addListener(details => {
-		if (details.reason === 'install') {
-			chrome.runtime.openOptionsPage();
-		}
-	});
-
-	chrome.browserAction.onClicked.addListener(tab => {
-		const tabUrl = window.GitHubNotify.getTabUrl();
-
-		// request optional permissions the 1rst time
-		if (window.GitHubNotify.settings.get('tabs_permission') === undefined) {
-			window.GitHubNotify.requestPermission('tabs').then(granted => {
-				window.GitHubNotify.settings.set('tabs_permission', granted);
-				openTab(tabUrl, tab);
-			});
-		} else {
-			openTab(tabUrl, tab);
-		}
-	});
+chrome.runtime.onInstalled.addListener(handleInstalled);
+chrome.browserAction.onClicked.addListener(handleBrowserActionClick);
```

Comment:
You sometimes use `window.chrome` and sometimes `chrome.` I think we should use only `chrome`.